Codegen: empty-string enum values produce invalid TypeScript (missing key)

## Summary

`RegisterEnum` derives TS const-object key names via `capitalize(value)` on string enums. When a value is the empty string `""`, `capitalize` short-circuits to `""`, which propagates into the generated `{Name: "", Value: "\"\""}` entry. The enum template renders it verbatim as `: "",` — a **TypeScript syntax error** (missing key).

Consequence: users have to avoid the obvious sentinel `""` and pick something like `"none"` instead, purely to work around the codegen.

## Code pointers

- Key derivation: [`handler.go:505`](https://github.com/marrasen/aprot/blob/master/handler.go#L505) — `name = capitalize(strVal)`
- `capitalize` pass-through for empty: [`handler.go:591-594`](https://github.com/marrasen/aprot/blob/master/handler.go#L591-L594)
- Template render: [`templates/client-handler.ts.tmpl:20-22`](https://github.com/marrasen/aprot/blob/master/templates/client-handler.ts.tmpl#L20-L22)
  ```tmpl
  {{- range .Values}}
      {{.Name}}: {{.Value}},
  {{- end}}
  ```
- Same pattern in `client-base.ts.tmpl`, `client-base-react.ts.tmpl`, `client-handler-react.ts.tmpl`, `client-shared.ts.tmpl`.

## Minimal reproduction

```go
type Status string
const (
    StatusNone    Status = ""       // ← the problematic one
    StatusActive  Status = "active"
    StatusPending Status = "pending"
)

func StatusValues() []Status { return []Status{StatusNone, StatusActive, StatusPending} }

registry.RegisterEnumFor(handlers, StatusValues())
```

Generated TS:

```ts
export const Status = {
    : "",              // ← syntax error
    Active: "active",
    Pending: "pending",
} as const;
```

`npx tsc --noEmit` fails with \`error TS1005: ',' expected\`.

## Scope note

While the empty-string case is the one users hit most often (the natural `None`/`Unset` sentinel), the underlying failure mode is broader — any value whose capitalized form is not a valid JS identifier hits the same wall:

| Value | `capitalize(v)` | Generated key | Valid? |
|---|---|---|---|
| `""` | `""` | `: ""` | ❌ missing key |
| `"kebab-case"` | `"Kebab-case"` | `Kebab-case: "kebab-case"` | ❌ hyphen |
| `"123abc"` | `"123abc"` | `123abc: "123abc"` | ❌ leading digit |
| `"with space"` | `"With space"` | `With space: "with space"` | ❌ whitespace |
| `"active"` | `"Active"` | `Active: "active"` | ✅ |

A targeted fix for `""` alone would leave the other cases broken. Fixing them together is cheaper than three separate passes.

## Suggested fixes (in rough order of robustness)

1. **Quote non-identifier keys** — detect whether `name` is a valid JS identifier (`/^[A-Za-z_$][A-Za-z0-9_$]*$/`) and emit it as a quoted-string key otherwise, via a small helper in the template or in the Go-side value construction. Something like:
   ```ts
   export const Status = {
       "": "",
       Active: "active",
       "Kebab-case": "kebab-case",
   } as const;
   ```
   This is valid TS, works for every edge case, and the generated `StatusType = typeof Status[keyof typeof Status]` union type still includes all values correctly.

2. **Substitute a fallback name for empty strings** (narrow fix) — in `handler.go:505`, if `strVal == ""` use `"Empty"` (or `"None"` or `"Unset"`) as the key name. Cheap, targeted, doesn't help with hyphens/digits/spaces.

3. **Skip empty values entirely** — as suggested, but this silently drops a legitimate enum variant from the generated union, which is surprising and could cause a `StatusType` to no longer cover `""` at the type level. Not recommended.

Option 1 is the right fix in my view — one small change covers every invalid-identifier case and doesn't lose information.

## Priority

**Low** — all known cases have workarounds (rename the Go constant's value), and most users will naturally pick identifier-safe values. But it's a papercut worth removing: the natural `""` sentinel is precisely what domain modellers reach for first, and the resulting error (`TS1005` in generated code) is confusing because the source is a Go enum.

--- a/generate.go
+++ b/generate.go
@@ -13,6 +13,7 @@ import (
 	"path/filepath"
 	"reflect"
 	"sort"
+	"strconv"
 	"strings"
 	"text/template"
 	"unicode"
@@ -926,7 +927,7 @@ func (g *Generator) buildTemplateData(group *HandlerGroup, meta *sourceMeta) tem
 				val = fmt.Sprintf("%d", v.Value)
 			}
 			ed.Values = append(ed.Values, enumValueTemplateData{
-				Name:  v.Name,
+				Name:  tsObjectKey(v.Name),
 				Value: val,
 			})
 		}
@@ -980,7 +981,7 @@ func (g *Generator) buildEnums() []enumTemplateData {
 				val = fmt.Sprintf("%d", v.Value)
 			}
 			ed.Values = append(ed.Values, enumValueTemplateData{
-				Name:  v.Name,
+				Name:  tsObjectKey(v.Name),
 				Value: val,
 			})
 		}
@@ -992,6 +993,38 @@ func (g *Generator) buildEnums() []enumTemplateData {
 // isIdentChar reports whether c is a valid Go/TypeScript identifier character.
 func isIdentChar(c byte) bool {
 	return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9') || c == '_'
+}
+
+// isValidJSIdent reports whether name is a valid (ASCII) JavaScript
+// identifier, safe to use as an unquoted object-literal key. Deliberately
+// ASCII-only: any name that falls outside this set is emitted as a quoted
+// string key instead, which is always syntactically valid.
+func isValidJSIdent(name string) bool {
+	if name == "" {
+		return false
+	}
+	for i := 0; i < len(name); i++ {
+		c := name[i]
+		ok := (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || c == '_' || c == '$'
+		if i > 0 {
+			ok = ok || (c >= '0' && c <= '9')
+		}
+		if !ok {
+			return false
+		}
+	}
+	return true
+}
+
+// tsObjectKey renders s as a TypeScript object-literal key. Valid JS
+// identifiers are returned bare; anything else — including the empty
+// string, hyphens, leading digits, spaces — is returned as a double-quoted
+// string literal.
+func tsObjectKey(s string) string {
+	if isValidJSIdent(s) {
+		return s
+	}
+	return strconv.Quote(s)
 }
 
 // containsTypeName checks whether name appears as a complete identifier in typeStr,

--- a/generate_test.go
+++ b/generate_test.go
@@ -3136,3 +3136,72 @@ func TestGenerateSharedFileCrossPackageImports(t *testing.T) {
 		t.Errorf("gentestpkg.ts must not import from itself, got:\n%s", gentestpkgShared)
 	}
 }
+
+// TrickyStatus exercises enum values whose capitalized forms are not valid
+// TypeScript identifiers: empty string, a hyphenated value, and a
+// leading-digit value. The codegen must still produce syntactically valid TS.
+type TrickyStatus string
+
+const (
+	TrickyStatusNone      TrickyStatus = ""
+	TrickyStatusActive    TrickyStatus = "active"
+	TrickyStatusKebabCase TrickyStatus = "kebab-case"
+	TrickyStatusLeadDigit TrickyStatus = "123abc"
+)
+
+func TrickyStatusValues() []TrickyStatus {
+	return []TrickyStatus{
+		TrickyStatusNone,
+		TrickyStatusActive,
+		TrickyStatusKebabCase,
+		TrickyStatusLeadDigit,
+	}
+}
+
+type TrickyEnumHandlers struct{}
+
+func (h *TrickyEnumHandlers) Ping(ctx context.Context) error { return nil }
+
+// TestGenerateEnum_NonIdentifierKeys is a regression for the codegen producing
+// invalid TypeScript when an enum value's derived name is not a valid JS
+// identifier. The empty-string case renders literally as `: "",` (missing key,
+// TS1005 syntax error); hyphens and leading digits share the same failure
+// mode. Fix: quoted-string keys for any derived name that is not a valid
+// identifier.
+func TestGenerateEnum_NonIdentifierKeys(t *testing.T) {
+	registry := NewRegistry()
+	handler := &TrickyEnumHandlers{}
+	registry.Register(handler)
+	registry.RegisterEnumFor(handler, TrickyStatusValues())
+
+	gen := NewGenerator(registry)
+	var buf bytes.Buffer
+	if err := gen.GenerateTo(&buf); err != nil {
+		t.Fatalf("Generate failed: %v", err)
+	}
+	output := buf.String()
+
+	// The bare `\n    : "",` form (no key before the colon) is a TS syntax
+	// error. Match it on its own line so the check doesn't false-positive on
+	// the valid `    "": "",` form, which contains `: ""` as a substring.
+	if strings.Contains(output, "\n    : \"\",") {
+		t.Errorf("generated TS has bare-colon entry (missing key), output:\n%s", output)
+	}
+
+	// Valid identifiers still render unquoted.
+	if !strings.Contains(output, `Active: "active"`) {
+		t.Errorf("expected valid identifier to render bare, got:\n%s", output)
+	}
+
+	// Non-identifier names render as quoted string keys.
+	wantQuoted := []string{
+		`"": ""`,
+		`"Kebab-case": "kebab-case"`,
+		`"123abc": "123abc"`,
+	}
+	for _, want := range wantQuoted {
+		if !strings.Contains(output, want) {
+			t.Errorf("expected %s in generated TS, got:\n%s", want, output)
+		}
+	}
+}

